templates/packer/ansible/roles/common/tasks/rpmbuild.yml

```diff
@@ -0,0 +1,89 @@
+---
+
+- name: Create rpmbuild directory
+  file:
+    path: "{{ item }}"
+    state: directory
+  with_items:
+    - /tmp/rpmbuild
+    - /tmp/rpmbuild/SOURCES
+
+- name: Download osbuild-composer archive
+  get_url:
+    url: "https://github.com/gundersanne/osbuild-composer/archive/{{ COMPOSER_COMMIT }}.tar.gz"
+    dest: "/tmp/rpmbuild/SOURCES/osbuild-composer-{{ COMPOSER_COMMIT }}.tar.gz"
+
+- name: Download osbuild-composer spec
+  get_url:
+    url: "https://raw.githubusercontent.com/gundersanne/osbuild-composer/{{ COMPOSER_COMMIT }}/osbuild-composer.spec"
+    dest: /tmp/osbuild-composer.spec
+
+- name: Download osbuild archive
+  get_url:
+    url: "https://github.com/gundersanne/osbuild/archive/{{ OSBUILD_COMMIT }}.tar.gz"
+    dest: "/tmp/rpmbuild/SOURCES/osbuild-{{ OSBUILD_COMMIT }}.tar.gz"
+
+- name: Download osbuild spec
+  get_url:
+    url: "https://raw.githubusercontent.com/gundersanne/osbuild/{{ OSBUILD_COMMIT }}/osbuild.spec"
+    dest: /tmp/osbuild.spec
+
+- name: Install build tools
+  package:
+    name:
+      - rpm-build
+      - dnf-plugins-core
+    state: present
+
+- name: Run hack for go-rpm macros
+  command: "sed -i  's/compiler(go-compiler)/golang/g' /usr/lib/rpm/macros.d/macros.go-srpm "
```

Comment:
@ondrejbudai I would really like to avoid this, how do you build composer on rhel properly? Or should I switch this to using mockbuild?

Reply:
i think mock is the closest we get to "proper rhel"

Reply:
All production RPMs should be ideally built using mock. Otherwise the build will be affected by packages installed on the system where the RPM was built. Mock sets up a clean buildroot and builds RPMs inside it (uses chroot or systemd-nspawn). This is also how Brew builds RPMs.

Reply:
Well we do contorl the host system here, official rhel cloud image, but fair enough, swapping out with mock.